Controller/Certificates/Index.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * ClassyLlama_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @copyright  Copyright (c) 2018 Avalara, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace ClassyLlama\AvaTax\Controller\Certificates;
+
+use Magento\Customer\Controller\RegistryConstants;
+use Magento\Framework\App\RequestInterface;
+
+class Index extends \Magento\Framework\App\Action\Action
+{
+    /**
+     * @var \Magento\Framework\View\Result\PageFactory
+     */
+    protected $resultPageFactory;
+
+    /**
+     * @var \Magento\Framework\Registry
+     */
+    protected $coreRegistry;
+
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    protected $customerSession;
+
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    protected $session;
+
+    /**
+     * @param \Magento\Framework\App\Action\Context      $context
+     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
+     * @param \Magento\Customer\Model\Session            $customerSession
+     * @param \Magento\Framework\Registry                $coreRegistry
+     * @param \Magento\Customer\Model\Session            $session
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
+        \Magento\Framework\View\Result\PageFactory $resultPageFactory,
+        \Magento\Customer\Model\Session $customerSession,
+        \Magento\Framework\Registry $coreRegistry,
+        \Magento\Customer\Model\Session $session
+    )
+    {
+        parent::__construct($context);
+
+        $this->resultPageFactory = $resultPageFactory;
+        $this->coreRegistry = $coreRegistry;
+        $this->customerSession = $customerSession;
+        $this->session = $session;
+    }
+
+    /**
+     * @param RequestInterface $request
+     *
+     * @return \Magento\Framework\App\ResponseInterface
+     * @throws \Magento\Framework\Exception\NotFoundException
+     */
+    public function dispatch(RequestInterface $request)
+    {
+        if (!$this->session->authenticate()) {
+            $this->_actionFlag->set('', self::FLAG_NO_DISPATCH, true);
+        }
+
+        return parent::dispatch($request);
+    }
+
+    /**
+     * @return \Magento\Framework\View\Result\Page
+     */
+    public function execute()
+    {
+        // Add the ID to the registry to allow for reusability with admin block
+        $this->coreRegistry->register(
```

Comment:
@LordZardeck - Another area where the clarity of the code by the direct dependence on the admin block.  It's not clear to me without hunting through the admin block class why it would be legitimate to set this registry value to 1 (or, more accurately, why a valid customer ID is not required in this context).  This controller needs to work in a way that doesn't require this.